schemachange/cli.py

```diff
@@ -271,11 +271,24 @@ def authenticate(self):
         warnings.warn(_warn_password, DeprecationWarning)
         snowflake_password = os.getenv("SNOWSQL_PWD")
 
+    # if password
     if snowflake_password:
```

Comment:
To keep style with the pattern of elif blocks currently in cli.py, rather than nesting the okta logic within the "if password" test, I'd recommend changing that test to "if password and not http", and then changing the "elif http" test to "elif http and password" (making sure to set the password in the latter).

e.g.
`        if snowflake_password and not os.getenv("SNOWFLAKE_AUTHENTICATOR").lower()[:8] == 'https://':
`
and

```
        elif os.getenv("SNOWFLAKE_AUTHENTICATOR").lower()[:8] == 'https://' \
                and os.getenv("SNOWFLAKE_AUTHENTICATOR") and \
                os.getenv("SNOWFLAKE_PASSWORD") is not None \
                and os.getenv("SNOWFLAKE_PASSWORD"):
            okta = os.getenv("SNOWFLAKE_AUTHENTICATOR")
            self.conArgs['password'] = snowflake_password
            self.conArgs['authenticator'] = okta

```